### PR TITLE
[chore][receiver/splunkenterprise] Splunkent wire component

### PR DESCRIPTION
**Pull request by shalper2**
**Description:** Graduate splunkenterprise receiver component to alpha

**Link to tracking Issue:**

**Testing:** Performed `make otelcontribcol` and ran resulting binary with the following config:
```yaml
extensions:
    basicauth/indexer:
        client_auth:
            username: admin
            password: securityFirst
    basicauth/cluster_master:
        client_auth:
            username: admin
            password: securityFirst

receivers:
    splunkenterprise:
        indexer:
            auth:
              authenticator: basicauth/indexer
            endpoint: "https://localhost:8089/"
            timeout: 45s
        cluster_master:
            auth:
              authenticator: basicauth/cluster_master
            endpoint: "https://localhost:8089/"
            timeout: 45s

exporters:
  otlp:
    endpoint: 127.0.0.1:8000

service:
  extensions: [basicauth/indexer, basicauth/cluster_master]
  pipelines:
    metrics:
      receivers: [splunkenterprise]
      exporters: [otlp]
```

and received the following output:
```
sh ~> ./otelcontribcol_linux_amd64 --config=file:config.yaml
2024-05-08T17:34:33.032-0500 info service@v0.100.0/service.go:102 Setting up own telemetry...
2024-05-08T17:34:33.032-0500 info service@v0.100.0/telemetry.go:103 Serving metrics {"address": ":8888", "level": "Normal"}
2024-05-08T17:34:33.032-0500 info receiver@v0.100.0/receiver.go:310 Development component. May change in the future. {"kind": "receiver", "name": "splunkenterprise", "data_type": "metrics"}
2024-05-08T17:34:33.033-0500 info service@v0.100.0/service.go:169 Starting otelcontribcol... {"Version": "0.100.0-dev", "NumCPU": 16}
2024-05-08T17:34:33.033-0500 info extensions/extensions.go:34 Starting extensions...
2024-05-08T17:34:33.033-0500 info extensions/extensions.go:37 Extension is starting... {"kind": "extension", "name": "basicauth/cluster_master"}
2024-05-08T17:34:33.033-0500 info extensions/extensions.go:52 Extension started. {"kind": "extension", "name": "basicauth/cluster_master"}
2024-05-08T17:34:33.033-0500 info extensions/extensions.go:37 Extension is starting... {"kind": "extension", "name": "basicauth/indexer"}
2024-05-08T17:34:33.033-0500 info extensions/extensions.go:52 Extension started. {"kind": "extension", "name": "basicauth/indexer"}
2024-05-08T17:34:33.033-0500 info service@v0.100.0/service.go:195 Everything is ready. Begin running and processing data.
```

indicating that the collector was able to successfully start with the component configured.

**Documentation:** Documentation was updated to indicate change in status from development to alpha

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -151,6 +151,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snmpreceiver v0.100.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver v0.100.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/solacereceiver v0.100.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkenterprisereceiver v0.0.0-00010101000000-000000000000
```
Comment (crobert-1) [inline go.mod:154]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkenterprisereceiver v0.100.0
```

File: cmd/configschema/go.mod [hunk 2]
```diff
@@ -580,6 +580,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/simpleprometheusreceiver v0.100.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snmpreceiver v0.100.0 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snowflakereceiver v0.100.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkenterprisereceiver v0.0.0-00010101000000-000000000000 // indirect
```
Comment (crobert-1) [inline cmd/configschema/go.mod:583]:
```suggestion
	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/splunkenterprisereceiver v0.100.0 // indirect
```
The replace overrides whatever version number is here, so we want to specify the latest release like all of the other components here (even though it wasn't in the given release).

**Comment by crobert-1** [comment]
To complete the move to `alpha` you'll want to update the stability in the component's `metadata.yaml`.

**Comment by shalper2** [comment]
sorry that was silly

**Review by crobert-1 — APPROVED** [review]
LGTM! Just in case you aren't aware, if you want this to be included in the released binary of `contrib`, you'll need to submit a PR adding it to the [releases repo](https://github.com/open-telemetry/opentelemetry-collector-releases). This is a recent example PR that you can generally follow: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/531

**Comment by crobert-1** [comment]
I ended up submitting a [PR for this receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32956) in parallel to add a header to its README. Since it displays the component's stability, can you rebase and run `make generate` to properly reflect your updates here?

**Comment by shalper2** [comment]
sorry about all the commits -- the rebase really rocked the boat

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**